source/daplink/drag-n-drop/file_stream.c

```diff
@@ -364,3 +373,50 @@ static error_t close_hex(void *state)
     status = flash_decoder_close();
     return status;
 }
+
+/* UF2 file processing */
+
+static bool detect_uf2(const uint8_t *data, uint32_t size)
+{
+    return 1 == validate_uf2block(data, size);
+}
+
+static error_t open_uf2(void *state)
+{
+    error_t status;
+    status = flash_decoder_open();
+    return status;
+}
+
+static error_t write_uf2(void *state, const uint8_t *data, uint32_t size)
+{
+    error_t status;
+    uint32_t start_addr;
+    const flash_intf_t *flash_intf;
+    const UF2_Block *block;
+
+    if (1 != validate_uf2block(data, size)) {
```

Comment:
This code assumes the calls to `write_uf2()` are one sector (512 bytes) long. Looking at `file_data_handler()` in `source/daplink/drag-n-drop/vfs_manager.c`, it looks like it could be multiple sectors (though I am not sure this can actually happen).
There is also the CMSIS-DAP / WebUSB commands (see `source/daplink/cmsis-dap/DAP_vendor.c`) for flashing files, because it works over the HID endpoint will (once framing has been removed) write about 40 bytes at a time.
We might need a buffer (the `shared_state_t` variant from hex files already has a 256 bytes buffer).